functions/helpFunctions.tsx

```diff
@@ -191,3 +193,128 @@ export function chooseCorrelatedAssessment(
+            maxPoints:
+              candidate.result.ext_inspera_questions[taskNum]
+                .ext_inspera_maxQuestionScore,
+            taskNumber: taskNum + 1,
+          });
+        });
+
+        //add to localstorage
+        localStorage.setItem(key, JSON.stringify(dataToBeUploaded));
+      }
+    });
+  }
+}
+
+export function getApprovedAssessments(taskNum: string): ApprovalType[] {
+  const approvedKey: string = taskNum + '_approved';
+  let approvedAssessments: ApprovalType[] = [];
+  if (typeof window !== 'undefined') {
+    if (approvedKey in localStorage) {
+      approvedAssessments = JSON.parse(
+        localStorage.getItem(approvedKey) as string
+      );
+    }
+  }
+  return approvedAssessments;
+}
+
+export function getRawAssessments(taskNum: string): AnswerType[] {
+  const rawKey: string = taskNum + '_data';
+  let rawAssessments: AnswerType[] = [];
+  if (typeof window !== 'undefined') {
+    if (rawKey in localStorage) {
+      rawAssessments = JSON.parse(localStorage.getItem(rawKey) as string);
+    }
+  }
+  return rawAssessments;
+}
+
+export function getStartedAssessments(taskNum: string): AssessmentType[] {
+  const startedKey: string = taskNum + '_assessments';
+  let startedAssessments: AssessmentType[] = [];
+  if (typeof window !== 'undefined') {
+    if (startedKey in localStorage) {
+      startedAssessments = JSON.parse(
+        localStorage.getItem(startedKey) as string
+      );
+    }
+  }
+  return startedAssessments;
+}
+
+export function getAssessmentData(taskNum: string) {
+  // get the different assessments from localstorage
+  const raw: AnswerType[] = getRawAssessments(taskNum).slice(0, 10);
+  const started: AssessmentType[] = getStartedAssessments(taskNum);
+  const approved: ApprovalType[] = getApprovedAssessments(taskNum);
+
+  // data to be returned
+  let assessmentData: AssessmentType[] = [];
+
+  // all answers have been assessed and approved, use approved data
+  if (approved.length != 0) {
+    assessmentData = approved;
+  }
+  // no assessments have been made for current task, use raw data
+  else if (started.length == 0) {
+    // convert to AssessmentType
+    assessmentData = raw.map((answer: AnswerType) => ({
+      score: '',
+      isFlagged: false,
+      ...answer,
+    }));
+  }
+
+  // assessments have started for this task, but there are still some remaining answers
+  // need to filter out those that have been assessed
+  // filter on candidateId
+  else {
+    const remainingAnswers = getDifference(raw, started);
+
+    // no remaining answers left, but not approved. This is for the back button in the approval page
+    if (remainingAnswers.length == 0 && approved.length == 0) {
+      assessmentData = started;
+    } else {
+      const p = remainingAnswers.map((answer: AnswerType) => ({
+        score: '',
+        isFlagged: false,
+        ...answer,
+      }));
+      assessmentData = p;
+    }
+  }
```

Comment:
This excludes the reAssessments...
Because the reAssessments aren't saved to localstorage.
I'll make a bug on it and fix it later.
Probably gonna add an `isInconsistant` boolean , much like the `isFlagged`